modules/core/include/opencv2/core/utils/filesystem.private.hpp

```diff
@@ -16,8 +16,8 @@
 #      define OPENCV_HAVE_FILESYSTEM_SUPPORT 1
 #  elif defined(__APPLE__)
 #    include <TargetConditionals.h>
-#    if (defined(TARGET_OS_OSX) && TARGET_OS_OSX) || (!defined(TARGET_OS_OSX) && !TARGET_OS_IPHONE)
-#      define OPENCV_HAVE_FILESYSTEM_SUPPORT 1 // OSX only
+#    if (defined(TARGET_OS_OSX) && TARGET_OS_OSX) || (defined(TARGET_OS_IOS) && TARGET_OS_IOS)
+#      define OPENCV_HAVE_FILESYSTEM_SUPPORT 1 // OSX, iOS only
```

Comment:
This condition is from 3.4 branch: https://github.com/opencv/opencv/blame/3.4/modules/core/include/opencv2/core/utils/filesystem.private.hpp#L19

relates #10018  (before any of 4.x releases)

---

What functionality you want to use on iOS?

Reply:
We found ::fs:: , and (strangely) some parts of videoio not working.

Opencv builds and works fine in our (limited) test cases for ios, macos when OPENCV_HAVE_FILESYSTEM_SUPPORT=1.

Reply:
> `::fs::`

This API should be considered as internal.
There is no goal to support FS utilities through Computer Vision Library.
Some of these utilities are available through C++17 filesystem features, consider using them instead: https://en.cppreference.com/w/cpp/filesystem

---

> some parts of videoio not working

Do you mean a problem similar to #19910?

Reply:
> `::fs::` This API should be considered as internal.
I agree. I have checked and we do not use ::fs on ios.
As a note, the whole namespace is listed and documented in the documentation, without any mentions of being internal.

We use cv::glob() (core/utils.hpp included by default with the core module) extensively, i have checked the  [PR](https://github.com/opencv/opencv/pull/20010/files#diff-75b7ddefa1f6e5cf61410a883f56c960ab9771bff45e3a1a0f2f8d225615b559) and it modifies this function as well

>videoio
No, we had an explicit assert/error mentioning the filesystem being disabled. The only place i found skimming through the module is loadPlugin()/getPluginCandidates()
We call cv::VideoCapture/open()